src/lib.rs

```diff
@@ -331,6 +335,29 @@ pub fn ciff_to_pisa(input: &Path, output: &Path) -> Result<()> {
         reorder_pisa_index(&index_paths)?;
     }
 
+    if generate_lexicons {
+        eprintln!("Generating the document and term lexicons...");
+
+        // Need to flush the document identifiers
+        trecids.flush()?;
+
+        let termlex: PayloadVector = std::fs::read_to_string(&index_paths.terms)?
+            .trim()
+            .split('\n')
+            .map(str::to_string)
+            .collect();
+        let mut lex_path = BufWriter::new(File::create(&index_paths.termlex)?);
+        termlex.write(&mut lex_path)?;
```

Comment:
Since we're using this code twice, it might be good to create a `build_lexicon` function (or whatever you feel the good name would be) that takes input txt file and output lexicon path, and does all this: reads, creates vector, writes.

We can actually add it in `payload_vector` module, and rewrite the `test_write` test slightly to use it. We'd need to first write the txt file instead of reading it from a file that's in git because of newlines problem on windows; in that case we can remove that file altogether I think; you could also rewrite this code to not explicitly use `\n` as separator, e.g., use [`lines`](https://doc.rust-lang.org/std/io/trait.BufRead.html#method.lines) on `BufReader` instance).

I think this might work, though I'm typing it off the top of my head:

```rust
let termlex = BufReader::new(File::open(input)?)
    .lines()
    .collect::<Result<PayloadVector, _>>()?;
```

Reply:
Agreed. I have worked the function in but have not yet updated the test case.